### ISSUE TITLE
System.Drawing.Common is no longer usable from < netcoreapp3.1 projects

### ISSUE THREAD

**Issue by koszeggy** [dotnet/runtime#59032]
### Description

Consider a class library with the following .csproj ([live project example](https://github.com/koszeggy/KGySoft.Drawing)):

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp2.0;netcoreapp3.0;net5.0</TargetFrameworks>
  </PropertyGroup>

  <ItemGroup>
    <!--<PackageReference Include="System.Drawing.Common" Version="5.0.2" />-->
    <PackageReference Include="System.Drawing.Common" Version="6.0.0-preview.7.21377.19" />
  </ItemGroup>

</Project>
```

This builds successfully with referencing v5.0.2 of `System.Drawing.Common` but after upgrading to .NET 6 Preview 7 it fails for targets `netcoreapp2.0` and `netcoreapp3.0` saying:

    System.Drawing.Common.targets(4,5): error : System.Drawing.Common doesn't support netcoreapp2.0. Consider updating your TargetFramework to netcoreapp3.1 or later.
    System.Drawing.Common.targets(4,5): error : System.Drawing.Common doesn't support netcoreapp3.0. Consider updating your TargetFramework to netcoreapp3.1 or later.

⚠️ **Please note that using `netstandard2.0` or `netstandard2.1` still works; however, these targets are ignored by a project targeting .NET Core 2.0/3.0** (and an application cannot target .NET Standard anyway) **because the referenced package has at least one .NET Core target, whose required version is now too high.**

### Configuration

* Windows 10
* Visual Studio 2019 16.11.2
* Installed SDK versions:
  * 2.1.526
  * 2.1.818
  * 2.2.401
  * 3.0.100
  * 3.1.412
  * 5.0.104
  * 5.0.206
  * 5.0.303
  * 5.0.400
  * 6.0.100-preview.7.21379.14

### Regression?

Yes. It seems that .NET Core 3.0 was lastly supported in [6.0.0-preview.5.21301.5](https://www.nuget.org/packages/System.Drawing.Common/6.0.0-preview.5.21301.5), whereas .NET Core 2.0 support existed till [6.0.0-preview.4.21253.7](https://www.nuget.org/packages/System.Drawing.Common/6.0.0-preview.4.21253.7).

### Other information

Unfortunately if the package contains at least one .NET Core target, then the compatible .NET Standard targets will be ignored even if they are actually supported. At least the `netcoreapp2.0` target must be added again to fix the issue even if that binary would be essentially the same as the `netstandard2.0` one.

**Comment by koszeggy** [comment]
Btw, a similar issue applies for [`System.Windows.Extensions`](https://www.nuget.org/packages/System.Windows.Extensions/6.0.0-preview.7.21377.19), which contains the `ImageConverter` class, for example. The official [compatibility table](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.imageconverter?view=net-5.0#applies-to) still says that `ImageConverter` is available in .NET Core 3.0, but it's no longer true starting with version  6.0.0-preview.6.21352.12.

Should I file another issue or can these be fixed together?

**Comment by filipnavara** [comment]
Note that both .NET Core 2 and .NET Core 3.0 went out of support. You can still target them but you cannot use the new version of System.Drawing.Common.

**Comment by koszeggy** [comment]
> Note that both .NET Core 2.1 and .NET Core 3.0 went out of support.

@filipnavara: This reasoning is valid for an end-user application that can (and should) regularly upgrade the versions, but a class library always supports the lowest possible versions. Since when are versions removed from the [`Applies To` tables](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.imageconverter?view=net-5.0#applies-to)? AFAIK this has never been a practice so far. 

**Note:** .NET Standard 2.0 is still supported by the package, which then makes no sense now. As the package has both .NET Core and .NET Framework targets, in an actual application the standard version never will be picked. Isn't it reasonable to support the lowest .NET 2.0 Standard compliant actual targets? If end users try to use an obsolete version, they will face the `NETSDK1138` warning anyway. Let's leave its handling to them, but let's not break all class libraries, especially if the solution is so simple.

**Comment by filipnavara** [comment]
> Since when are versions removed from the Applies To tables? AFAIK this has never been a practice so far.

They are not removed. If you target unsupported versions you are free to use older version of the System.Drawing.Common NuGet. There's simply no guarantee that newer version of the out-of-box assemblies would still target frameworks that went out of the support cycle.

cc @ViktorHofer

**Comment by koszeggy** [comment]
> .NET Framework 4.8 is still supported and it only supports .NET Standard 2.0.

But the package contains also .NET Framework 4.6.1 so when targeting 4.8, _not_ the .NET Standard 2.0 will be picked but the Framework binaries. If it was true I could happily continue targeting .NET Core 2.0/3.0 because they also could pick the .NET Standard 2.0 binaries (this is what I tried to explain, I hope it is clearer now). For .NET Framework this is not a problem, 4.6.1 is the lowest .NET 2.0 Standard compatible version anyway.

> You can even target different NuGet versions [...]

While I'm aware of this I would hesitate to introduce a possible .dll hell in my libraries.

**Comment by danmoseley** [comment]
@ViktorHofer

**Comment by ViktorHofer** [comment]
@koszeggy please see the [announcement issue](https://github.com/dotnet/announcements/issues/190) that lists the supported target frameworks for .NET 6. 

The reason why packages like System.Drawing.Common are blocked on < netcoreapp3.1 even though they have a netstandard2.0 asset is that in many cases the .NETStandard asset just contains type forwards and not the actual implementation.
You can see that by downloading the S.D.Common nuget package and opening the netstandard2.0 asset in i.e. ilspy.

Overall, packages produced by dotnet/runtime starting with .NET 6 only support the announced tfms. Ie for .NET 7, net461 won't be supported anymore: https://devblogs.microsoft.com/dotnet/net-framework-4-5-2-4-6-4-6-1-will-reach-end-of-support-on-april-26-2022/

**Comment by maryamariyan** [comment]
deferring to @ViktorHofer on how to move forward with this issue.
